File: lib/models/task_page_result.dart
import 'task.dart';

/// 任务分页返回（Unify API）
class TaskPageResult {
  final int totalPages;
  final int totalElements;
  final int numberOfElements;
  final int size;
  final int number; // 当前页号（后端返回，0 开始）
  final List<Task> content;

  const TaskPageResult({
    required this.totalPages,
    required this.totalElements,
    required this.numberOfElements,
    required this.size,
    required this.number,
    required this.content,
  });
}

File: src/ChatDesktop.Core/Dto/TaskPageResult.cs
using ChatDesktop.Core.Models;

namespace ChatDesktop.Core.Dto;

/// <summary>
/// 任务分页结果
/// </summary>
public sealed class TaskPageResult
{
    public int TotalPages { get; set; }
    public int TotalElements { get; set; }
    public int NumberOfElements { get; set; }
    public int Size { get; set; }
    public int Number { get; set; }
    public IReadOnlyList<TaskItem> Content { get; set; } = Array.Empty<TaskItem>();
}
